inc/Smartling/Helpers/PlaceholderHelper.php

```diff
@@ -12,6 +12,11 @@ public function addPlaceholders(string $string): string
         return self::SMARTLING_PLACEHOLDER_MASK_START . $string . self::SMARTLING_PLACEHOLDER_MASK_END;
     }
 
+    public function hasPlaceholders(string $string): bool
+    {
+        return preg_match('~' . self::SMARTLING_PLACEHOLDER_MASK_START . '.+' . self::SMARTLING_PLACEHOLDER_MASK_END . '~', $string);
```

Comment:
Use non non-greedy pattern. Test this string `#sl-start#test1#sl-end# test2 #sl-start#test3#sl-end#`
```suggestion
        return preg_match('~' . self::SMARTLING_PLACEHOLDER_MASK_START . '.+?' . self::SMARTLING_PLACEHOLDER_MASK_END . '~', $string);
```

Reply:
We only detect placeholder strings without matching, I've added a test, and it passes as-is